### PR TITLE
Simulation changes to support foot dot implementation

### PR DESCRIPTION
**Pull request by ucr-bruce-wang**

### CODE REVIEW

File: source/extensions/omni.isaac.lab/omni/isaac/lab/envs/mdp/observations.py
Comment (wwangucr) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/envs/mdp/observations.py:1]: we could move the implementation to the ucr_rl instead

File: source/extensions/omni.isaac.lab/omni/isaac/lab/envs/mdp/observations.py [hunk 1]
```diff
@@ -173,6 +173,74 @@ def height_scan(env: ManagerBasedEnv, sensor_cfg: SceneEntityCfg, offset: float
     return sensor.data.pos_w[:, 2].unsqueeze(1) - sensor.data.ray_hits_w[..., 2] - offset
 
 
+def foot_scan(env: ManagerBasedEnv, sensor_cfg: SceneEntityCfg) -> torch.Tensor:
+    """Foot scan works to find the absolute distance between the foot and ground through transformation calculations"""
+    
+    foot_ray_data = env.scene.sensors[sensor_cfg.name].data
+
+    def _quat_to_mat(q: torch.Tensor) -> torch.Tensor:
```
Comment (wwangucr) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/envs/mdp/observations.py:181]: Let's avoid duplication, there is another one in the reward function

File: source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/ray_caster.py [hunk 2]
```diff
@@ -214,6 +214,9 @@ def _initialize_rays_impl(self):
         self._data.quat_w = torch.zeros(self._view.count, 4, device=self._device)
         self._data.ray_hits_w = torch.zeros(self._view.count, self.num_rays, 3, device=self._device)
 
+        self._data.ray_starts = self.ray_starts.clone()
```
Comment (wwangucr) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/ray_caster.py:217]: My understanding is these value are set once during initialization, if that's true we don't need to put them in the self._data, we could access them directly through the ray caster class?
